profile/src/main/java/rdx/works/profile/cloudbackup/data/DriveClient.kt

```diff
@@ -251,7 +251,14 @@ class DriveClientImpl @Inject constructor(
     private suspend fun <T> Result<T>.mapDriveError(): Result<T> = mapError { error ->
         val mappedError = when (error) {
             is BackupServiceException -> error
-            is GoogleAuthIOException -> BackupServiceException.UnauthorizedException
+            is GoogleAuthIOException -> {
+                // User either revoked access to Google Drive from inside Google Drive settings
+                // or deleted the hidden app data (backup files)
+                // or both (revoke and delete hidden app data).
+                // Therefore remove last cloud backup event.
+                preferencesManager.removeLastCloudBackupEvent()
+                BackupServiceException.UnauthorizedException
+            }
```

Comment:
I am not sure that the checks are correct here. `GoogleAuthIOException` is only thrown when you have revoked access from drive. The deleting part is just coincidental. 

Also if we have to remove the last cloud backup event on such, we need to also to do that when we receive the BackupServiceException.UnAuthorizedException from `getDrive()` function. It seems confusing now but this has to do with how the sign in client is in another class from DriveClient.

Reply:
>I am not sure that the checks are correct here. GoogleAuthIOException is only thrown when you have revoked access from drive. The deleting part is just coincidental.

what is actually not correct is my comment there. Because if the user deletes only the app data from the Drive settings we will get a not found error and not GoogleAuthIOException:
<img width="1198" alt="Screenshot 2024-06-20 at 14 40 01" src="https://github.com/radixdlt/babylon-wallet-android/assets/118305718/8e7bcf8d-02fa-4784-af54-072d4df63712">
and we already removed the last cloud back up event there.


> Also if we have to remove the last cloud backup event on such, we need to also to do that when we receive the BackupServiceException.UnAuthorizedException from getDrive() function. It seems confusing now but this has to do with how the sign in client is in another class from DriveClient.

This is not necessary because the sign in flow is a completely different flow, and it takes place in:
- new wallet creation: so no need to remove the last cloud backup event
- restore wallet from cloud backup:  so no need to remove the last cloud backup event
- if the user decides to enable the cloud backup from the Backup screen:  so no need to remove the last cloud backup event
Am I missing something? 👀